cargo-pgrx/src/command/install.rs

```diff
@@ -161,10 +161,14 @@ pub(crate) fn install_extension(
         };
         dest.push(format!("{}.so", so_name));
 
-        if cfg!(target_os = "macos") {
+        if cfg!(target_os = "macos") || cfg!(target_os = "linux") {
```

Comment:
It seems to me we should just always do it this way, keeping the comments as to why.  macos and linux are our two primary target platforms and if they both exhibit similar problems with overwriting the existing .so, I'm sure other platforms do too.